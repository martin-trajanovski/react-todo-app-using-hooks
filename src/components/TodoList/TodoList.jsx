import React, { useReducer } from 'react';
import TodoItem from '../TodoItem/TodoItem';
import AddTodo from '../AddTodo/AddTodo';
import { initialState, todoReducer } from '../../reducers/todoReducer';

export default function TodoList() {
  const [state, dispatch] = useReducer(todoReducer, initialState);
  const { todos } = state;

  return (
    <div className="todo-list">
      <AddTodo
        addTodo={todo => dispatch({ type: 'ADD_TODO', payload: todo })}
      />
      {todos.map(todo => (
        <TodoItem
          key={todo.id}
          todo={todo}
          toggleTodo={() =>
            dispatch({ type: 'TOGGLE_TODO', payload: { id: todo.id } })
          }
          removeTodo={() =>
            dispatch({ type: 'REMOVE_TODO', payload: { id: todo.id } })
          }
        />
      ))}
    </div>
  );
}
