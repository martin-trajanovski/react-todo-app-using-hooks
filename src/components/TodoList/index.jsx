import React from 'react';
import { useSelector, useDispatch } from 'react-redux';
import TodoItem from '../TodoItem';
import { toggleTodo, removeTodo } from '../../actions';

export default function TodoList() {
  const todos = useSelector(state => state.todos);
  const dispatch = useDispatch();

  return (
    <div className="todo-list">
      {todos.map(todo => (
        <TodoItem
          key={todo.id}
          todo={todo}
          toggleTodo={() => dispatch(toggleTodo(todo.id))}
          removeTodo={() => dispatch(removeTodo(todo.id))}
        />
      ))}
    </div>
  );
}
