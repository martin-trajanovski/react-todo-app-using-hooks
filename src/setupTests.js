// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import '@testing-library/jest-dom/extend-expect';
import React from 'react';
import { render } from '@testing-library/react';
import { createStore } from 'redux';
import { Provider } from 'react-redux';

import rootReducer from './reducers';
import registerIcons from './registerIcons';

registerIcons();

export const renderWithRedux = (
  ui,
  { initialState, store = createStore(rootReducer, initialState) } = {}
) => {
  return {
    ...render(<Provider store={store}>{ui}</Provider>),
    // adding `store` to the returned utilities to allow us
    // to reference it in our tests (just try to avoid using
    // this to test implementation details).
    wrappedWithProvider: <Provider store={store}>{ui}</Provider>,
    store,
  };
};
