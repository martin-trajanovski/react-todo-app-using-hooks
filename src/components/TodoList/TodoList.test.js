import React from 'react';
import renderer from 'react-test-renderer';
import TodoList from '.';
import { renderWithRedux } from '../../setupTests';

test('should display todos correctly', () => {
  const { container } = renderWithRedux(<TodoList />);
  const todoList = container.querySelectorAll('.todo-item');

  expect(todoList.length).toEqual(1);
});

test('should render todos correctly', () => {
  const { wrappedWithProvider } = renderWithRedux(<TodoList />);
  const tree = renderer.create(wrappedWithProvider).toJSON();

  expect(tree).toMatchSnapshot();
});
