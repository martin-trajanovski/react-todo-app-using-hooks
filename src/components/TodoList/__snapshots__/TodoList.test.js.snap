// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render todos correctly 1`] = `
<div
  className="todo-list"
>
  <div
    className="todo-item"
  >
    <div
      className="alert alert-primary"
      role="alert"
    >
      <span
        className="todo-item-text"
        onClick={[Function]}
      >
        Build really cool todo app
      </span>
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-trash fa-w-14 float-right"
        color="#dc3545"
        data-icon="trash"
        data-prefix="fas"
        focusable="false"
        onClick={[Function]}
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16zM53.2 467a48 48 0 0 0 47.9 45h245.8a48 48 0 0 0 47.9-45L416 128H32z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </div>
  </div>
</div>
`;
